python/ray/serve/_private/deployment_state.py

```diff
@@ -2717,13 +2724,10 @@ def update(self) -> bool:
             and len(ds._replicas.get()) == ds.target_num_replicas
             for ds in self._deployment_states.values()
         )
-        draining_nodes = self._cluster_node_info_cache.get_draining_nodes()
-        if (
-            not draining_nodes
-            and RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY
-            and allow_active_compaction
-        ):
-            compact_opp = self._deployment_scheduler.detect_compact_opportunities()
+        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY:
+            compact_opp = self._deployment_scheduler.detect_compact_opportunities(
+                allow_active_compaction=allow_active_compaction
+            )
```

Comment:
```suggestion
        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY:
            node_info: CompactingNodeInfo = self._deployment_scheduler.choose_node_to_compact(
                allow_active_compaction=allow_active_compaction
            )
```

Reply:
The flag passed in is also a bit strange. Maybe:
```suggestion
        if RAY_SERVE_USE_COMPACT_SCHEDULING_STRATEGY:
            compact_opp = self._deployment_scheduler.detect_compact_opportunities(
                compact_new_node=allow_active_compaction
            )
```